doc/mclag/MCLAG_Enhancements_HLD.md

```diff
@@ -0,0 +1,805 @@
+
+### 2.1.7 Performance improvements
+
+The following improvements are made in this enhancement: 
+
+- Convert FDB and VLAN related linked list to binary trees to improve scaling performance
+- When MCLAG node learns a new MAC address via ICCP from peer MCLAG node learned over remote orphan port, it sends a age notification back to peer to indicate MAC is locally aged, this update is not necessary as MAC not learned locally. Avoid sending the local age notification back to peer MCLAG node each time a MAC address is learned from peer.  
+
+### 2.1.8 Isolation group support
+
+To control the BUM traffic towards the MHD bridge-port isolation groups is used. The isolation group is applied on the ingress side of peer-link port to filter traffic towards the MHD to avoid duplicate traffic received at MHD.
+
+Existing ACL based support will continue to support. 
+
+### 2.1.9 Unique IP enhancements
+
+- Current design mandates configuring same IP address on the VLAN interface for each VLAN of which an MCLAG is a member. 
+- L3 protocols such as BGP, BFD, VRRP etc cannot be configured on the VLAN interface associated with MCLAG interface because such protocols cannot address the individual routers. 
+- A new configuration command is introduced for allowing separate per-peer IP configuration on the VLAN interfaces associated with MCLAG interface.
+- Both same IP configuration and unique IP configuration will co-exist based on the new configuration.  
+
+
+# 3 Design
+
+## 3.1 Overview
+New changes will be added on top of the existing MCLAG implementation. 
+Following diagram captures the high level module changes made for the MCLAG enhancements. 
+
+![Design overview diagram](images/DesignOverviewDiagram.png "Overview of module changes")
+
+**Figure 1: Overview of module changes**
+
+Below is the summary of the changes for each flow number mentioned in the above diagram.
+
+1. New CLI commands are introduced to configure MCLAG and its related attributes. Two new tables are added in CONFIG_DB to store the MCLAG configuration and mclag_interface configuration generated from CLI.
+2. MclagSyncd receives the  new MCLAG configuration updates.
+3. MclagSyncd receives new local (both static and dynamic) MAC updates from STATE_DB FDB_TABLE
+4. MclagSyncd sends new MCLAG configuration to ICCPd.
+5. ICCPd sends new remote static MAC updates, ICCP session and MCLAG remote interface state information updates to MclagSyncd.
+6. ICCPd sets new static MAC flag in FDB TLV for local static MAC addresses advertisement. 
+7. MclagSyncd updates ICCP session and MCLAG remote interface state information to STATE_DB FDB_TABLE and MCLAG_REMOTE_INTF_TABLE
+8. MclagSyncd updates MAC addresses learned from peer MCLAG node to new  MCLAG FDB table .
+9. FdbOrch registers for new MCLAG FDB table updates to process MAC updates from peer MCLAG node, ISOGRP Orch process new updates from MclagSyncd.
```

Comment:
As per flows numbering in the diagram, looks like the flow 9 covers ISOGRP update through STATE DB only. A separate flow should start from MCLAG FDB (APP DB) to FdbOrch.

Reply:
Thanks @akokhan: The diagram is overall highlevel design the detailed flows are covered in section 4. ISOGRP Orch specific processing is further covered in the flow diagram  4.2.